### PR TITLE
feat: [sc-32064] Change library saving to send LO cuid and version

### PR DESCRIPTION
**Pull request by Greek**
This change introduces new parameters to allow clark-client to fetch a specific LO in a user's library, rather than the whole library itself. 

In a past implementation of this feature, the `page` and `limit` parameters of fetching a user's library were not functional, despite existing in the client query params. `learning-object-service` would just return most, if not all, learning objects that was saved in a user's library.

Now that the `page` and `limit` params work as intended we also need a way to fetch a learning object that exists in the users library, because the LO we want may not be returned within the requested page or limit. This PR introduces the necessary client-side fixes to fetch specific LOs.

https://app.shortcut.com/clarkcan/story/32064/fix-saving-an-lo-to-a-user-s-library

### CODE REVIEW

**Review by johngonz0 — APPROVED** [review]
nice find

**Review by stef359 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]
Just paige's comment

File: src/app/core/library-module/library.service.ts [hunk 1]
```diff
@@ -39,7 +39,9 @@ export class LibraryService {
     // this.headers.append('Content-Type', 'application/json');
   }
 
-  async getLibrary(page?: number, limit?: number): Promise<{ cartItems: LearningObject[], lastPage: number }> {
+  async getLibrary(page?: number, limit?: number,
+                    learningObjectCuid?: string,
+                    version?: number): Promise<{ cartItems: LearningObject[], lastPage: number }> {
     this.updateUser();
```
[the remark this answers is not in the document]
Comment (Greek) [reply src/app/core/library-module/library.service.ts:45]: changed the params to accept an object

**Review by curlyLasagna — APPROVED** [review]

File: src/app/core/library-module/library.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pzalep1) [reply src/app/core/library-module/library.service.ts:45]: I think that change didn't push.

[the remark this answers is not in the document]
Comment (Greek) [reply src/app/core/library-module/library.service.ts]: pushed

**Review by pzalep1 — APPROVED** [review]

**Pull request merged**